docs/application/native/guides/machine-learning/machine-learning-inference.md

```diff
@@ -0,0 +1,350 @@
+      uintarray1[i][1] = i + 1;
+      uintarray1[i][2] = i + 3;
+      uintarray1[i][3] = i + 2;
+    }
+
+    status = ml_tensors_data_set_tensor_data (data, 0, uintarray1[0], 4);
+
+    /* setiing the policy of raw data pointer */
+    status = ml_pipeline_src_input_data (srchandle, data, ML_PIPELINE_BUF_POLICY_DO_NOT_FREE);
+  ```
+
+  After configuring it, release the handle is required.
+
+  ```bash
+    status = ml_pipeline_src_release_handle (srchandle);
+  ```
+
+* #### Sink <a name="sinkelement"></a>
+  In case of getting the output tensor raw data, the configuration of data sink element is required.
+  ```bash
+    gchar * pipeline = g_strdup ("videotestsrc num-buffers=3 ! videoconvert ! tensor_converter ! appsink name=sinkx");
+  ```
+
+  Through `ml_pipeline_sink_register ()`, `appsink` element with the name `sinkx` become reachable.
+  ```bash
+    int status;
+    ml_pipeline_h handle;
+    ml_pipeline_sink_h sinkhandle;
+
+    status = ml_pipeline_sink_register (handle, "sinkx", sink_callback, user_data, &sinkhandle);
+  ```
+
+  `sink_callback ()` is called whenever appsink, `sinkx`, receives data. with [`data`](#tensorsdata), [`info`](#tensorsinfo), we can get the output data easily, and `user_data` can be used in the callback.
+
+  ```bash
+    typedef void (*ml_pipeline_sink_cb) (const ml_tensors_data_h data, const ml_tensors_info_h info, void *user_data);
+  ```
+
+  After using it, the `sinkhandler` has to be released through `ml_pipeline_sink_unregister ()`
+  ```bash
+    status = ml_pipeline_sink_unregister (sinkhandle);
+  ```
+
+
+* #### Valve <a name="valveelement"></a>
+  When the stream of a pipeline needs to be controlled, this element is useful.
+
+  ```bash
+    gchar * pipeline = g_strdup ("videotestsrc is-live=true ! videoconvert ! videoscale ! video/x-raw,format=RGBx,width=16,height=16,framerate=10/1 ! tensor_converter ! queue ! valve name=valve1");
+    int status = ml_pipeline_construct (pipeline, NULL, NULL, &handle);
+  ```
+
+  `valve1` is the name of the valve of the pipeline, and it is opened as a default. This valve can be controlled with getting `ml_pipeline_valve_h`.
+
+  ```bash
+    ml_pipeline_h handle;
+    ml_pipeline_valve_h valve1;
+
+    status = ml_pipeline_valve_get_handle (handle, "valve1", &valve1);
+  ```
+
+  After start pipeline, the stream of a pipeline can be controlled with valve.
+  ```bash
+    status = ml_pipeline_start (handle);
+
+    status = ml_pipeline_valve_set_open (valve1, true); /* close */
+  ```
+
+  It can be opened with the same way.
+
+  ```bash
+    status = ml_pipeline_valve_set_open (valve1, true); /* open */
+  ```
+
+  Before destroying the pipeline, `ml_pipeline_valve_h` has to be released.
+
+  ```bash
+    status = ml_pipeline_valve_release_handle (valve1); /* release valve handle */
+  ```
+
+* #### Switch <a name="switchelement"></a>
+  When a pipeline consists of multiple branches, and only one of the branches has to be working, this element is useful.
```

Comment:
Please rephrase to:
The switch element is used when you need only one working branch from a pipeline that has multiple branches: